spec/models/concerns/searchable_spec.rb

```diff
@@ -0,0 +1,57 @@
+require 'rails_helper'
+
+RSpec.describe Searchable do
+
+  search_columns = [:data_type, :name]
+  let(:dummy_class) { class_for_search_columns(*search_columns) }
+
+  context "with searchable_on called" do
+    it "should add searching class methods for specified search_columns" do
+      dummy_class1 = class_for_search_columns(:name) 
+      dummy_class2 = class_for_search_columns(:type) 
+      dummy_class3 = class_for_search_columns(:name, :type) 
+
+      expect(dummy_class1).to respond_to(:where_name_contains)
+      expect(dummy_class2).to respond_to(:where_type_contains)
+      expect(dummy_class3).to respond_to(:where_name_contains)
+      expect(dummy_class3).to respond_to(:where_type_contains)
+    end
+  end
+
+
+  search_columns.each do |search_column|
+    describe ".where_#{search_column}_contains" do
+
+      let(:search_string) { 'o' }
+      subject { dummy_class.send("where_#{search_column}_contains", search_string) }
+
+      before do
+        dummy_class.create(name: 'Hallo', data_type: 'json')
+        dummy_class.create(name: 'Hallo2', data_type: 'json')
+        dummy_class.create(name: 'Ciao', data_type: 'html')
+        dummy_class.create(name: 'Tschüss', data_type: 'html')
+        dummy_class.create(name: 'Whatever', data_type: 'html')
+      end
+
+      context 'without search_string given' do
+        let(:search_string) { '' }
+        it { is_expected.to eq(dummy_class.all) }
+      end
+
+      context 'without search_string given' do
+        let(:search_string) { 'o' }
+        it { is_expected.to eq(dummy_class.where("#{search_column} LIKE '%#{search_string}%'")) }
```

Comment:
here you are testing with the exact code from the class singleton method, right?
so we are testing that the code does the same there and here. do we want this? i mean it is given, that the same code does the same?
i could think of a more "human" test additionally like "results in three hits" or something?

Reply:
Like a more static test?
Yeah, maybe that's better in this case.

Testing the same code with the same code is pretty senseless...